api.go

```diff
@@ -167,11 +175,14 @@ func (api *CacophonyAPI) authenticate() error {
 	if api.device.password == "" {
 		return errors.New("no password set")
 	}
-	payload, err := json.Marshal(map[string]string{
+
+	payload, err := json.Marshal(map[string]interface{}{
+		"deviceID":   api.device.id,
 		"devicename": api.device.name,
 		"groupname":  api.device.group,
 		"password":   api.device.password,
```

Comment:
Is the API server `authenticate_device` endpoint already able to support logins with just the device id? If so, perhaps don't send the group name and password if there's a device id available?

Reply:
password is always needed